conversations.go

```diff
@@ -0,0 +1,205 @@
+package main
+
+import (
+	"accounts-service/auth"
+	"accounts-service/models"
+	accountsv1 "accounts-service/protorepo/noted/accounts/v1"
+	"accounts-service/validators"
+	"context"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type conversationsAPI struct {
+	accountsv1.UnimplementedConversationsAPIServer
+
+	auth         auth.Service
+	logger       *zap.Logger
+	messageRepo  models.ConversationMessagesRepository
+	repo         models.ConversationsRepository
+	groupService accountsv1.GroupsAPIServer
+}
+
+var _ accountsv1.ConversationsAPIServer = &conversationsAPI{}
+
+func (server *conversationsAPI) CreateConversation(ctx context.Context, in *accountsv1.CreateConversationRequest) (*accountsv1.CreateConversationResponse, error) {
+	token, err := server.authenticate(ctx)
+	if err != nil {
+		return nil, status.Error(codes.Unauthenticated, err.Error())
+	}
+
+	accountId := token.UserID.String()
+
+	_, err = server.groupService.GetGroupMember(ctx, &accountsv1.GetGroupMemberRequest{GroupId: in.GroupId, AccountId: accountId})
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, "sender not in group")
+	}
+
+	err = validators.ValidateCreateConversationRequest(in)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
```

Comment:
Validation is done after call to database. (Check if it is the case in other endpoints too)